### PR TITLE
Flush databases by RPC

### PR DESCRIPTION
**Pull request by SergioDemianLerner**
This PR enables a new RPC method ("rsk_flush") that flushes all state databases to disk.
## Description

We add a new RPC command which calls the BlockChainFlusher class.
We also fix the flush method of DataSourceWithCache that previously didn't flushed the database (which wasn't a problem before because the databases used so far perform automatic flushing)
## Motivation and Context

There are several reasons it is useful to make sure the state of the node databases have been written to disk without interrupting the node:
* If one suspects that power will soon fail
* For fair comparisons between databases (RocksDB vs LevelDB) is it important to assure both have committed all changes to disk
* To safely backup the databases without interrupting the node
* To access the databases in realtime from another application without interrupting the node.
* To accurately compute the current size of the databases without interrupting the node.

## How Has This Been Tested?

By executing the rsk_flush RPC command by hand and following all the flush code flow in the debugger, and also by checking the state of the database.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

**Other information**:
There is no documentation for RPC commands in the code, so no documentation was added to the code.

If merged, I will contact the developer's site maintainer to update the corresponding page.

### CODE REVIEW

**Comment by Vovchyk** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/RskContext.java [hunk 1]
```diff
@@ -847,12 +848,34 @@ public synchronized RskModule getRskModule() {
...
+                    getBlockStore(),
+                    getReceiptStore(),
+                    getBlocksBloomStore(),
+                    getStateRootsStore());
+        }
+
+        return blockChainFlusher;
+    }
+
+    public Flusher getFlusher() {
```
Comment (illuque-iov) [inline rskj-core/src/main/java/co/rsk/RskContext.java:875]: Minor thing, but do we need this method if we only have the `BlockchainFlusher`? It's a bit confusing, it makes me thing we might have several and this method would "select" one.

**Review by illuque-iov — APPROVED** [review]

**Comment by Vovchyk** [comment]
pipeline:run

**Review by illuque-iov — APPROVED** [review]

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Review by illuque-iov — APPROVED** [review]

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Pull request merged**